app/modules/import-export-customization/assets/js/import/components/drop-zone.js

```diff
@@ -0,0 +1,303 @@
+
+		if ( e.dataTransfer.items && e.dataTransfer.items.length > 0 ) {
+			setIsDragOver( true );
+		}
+	};
+
+	const handleDragLeave = ( e ) => {
+		e.preventDefault();
+		e.stopPropagation();
+
+		setDragCounter( ( prev ) => prev - 1 );
+
+		if ( dragCounter <= 1 ) {
+			setIsDragOver( false );
+		}
+	};
+
+	const handleDragOver = ( e ) => {
+		e.preventDefault();
+		e.stopPropagation();
+	};
+
+	const handleDrop = ( e ) => {
+		e.preventDefault();
+		e.stopPropagation();
+
+		setIsDragOver( false );
+		setDragCounter( 0 );
+
+		if ( isLoading ) {
+			return;
+		}
+
+		const file = e.dataTransfer.files[ 0 ];
+
+		if ( file && isValidFileType( file.type, file.name ) ) {
+			onFileSelect( file, e, 'drop' );
+		} else {
+			onError( {
+				id: 'file_not_allowed',
+				message: __( 'This file type is not allowed', 'elementor' ),
+			} );
+		}
+	};
+
+	const handleFileInputChange = ( e ) => {
+		const file = e.target.files[ 0 ];
+		if ( ! file ) {
+			return;
+		}
+
+		if ( ! isValidFileType( file.type, file.name ) ) {
+			onError( {
+				id: 'file_not_allowed',
+				message: __( 'This file type is not allowed', 'elementor' ),
+			} );
+			return;
+		}
+
+		onFileSelect( file, e, 'browse' );
+
+		if ( onFileChoose ) {
+			onFileChoose( file );
+		}
+	};
+
+	const handleButtonClick = ( e ) => {
+		if ( onButtonClick ) {
+			onButtonClick( e );
+		}
+
+		if ( fileInputRef.current ) {
+			fileInputRef.current.click();
+		}
+	};
+
+	const getAcceptedFileTypes = () => {
+		const acceptTypes = [ ...filetypes ];
+		const validExtensions = getValidExtensions();
+
+		validExtensions.forEach( ( ext ) => {
+			acceptTypes.push( `.${ ext }` );
+		} );
+
+		return acceptTypes.join( ',' );
+	};
+
+	return (
+		<Box
+			className={ className }
+			onDragEnter={ handleDragEnter }
+			onDragLeave={ handleDragLeave }
+			onDragOver={ handleDragOver }
+			onDrop={ handleDrop }
+			data-testid="drop-zone"
+			{ ...props }
+		>
+			<Box
+				sx={ {
+					border: 2,
+					borderColor: isDragOver ? 'primary.main' : 'divider',
+					borderStyle: 'dashed',
+					backgroundColor: isDragOver ? 'action.hover' : 'background.default',
+					transition: 'all 0.2s ease-in-out',
+					cursor: isLoading ? 'not-allowed' : 'normal',
+					position: 'relative',
+					overflow: 'hidden',
+				} }
+			>
+				<Box
+					display="flex"
+					flexDirection="column"
+					alignItems="center"
+					justifyContent="center"
+					gap={ ( theme ) => theme.spacing( 1 ) }
+					p={ ( theme ) => theme.spacing( 6 ) }
+					minHeight={ ( theme ) => theme.spacing( 65 ) }
+				>
+					{ isLoading && (
+						<Box
+							position="absolute"
+							top={ 0 }
+							left={ 0 }
+							right={ 0 }
+							bottom={ 0 }
+							display="flex"
+							alignItems="center"
+							justifyContent="center"
+							bgcolor="action.hover"
+							zIndex={ 1 }
+							data-testid="loading-overlay"
+						>
+							<CircularProgress size={ 40 } data-testid="loading-spinner" />
+						</Box>
```

Comment:
Why should it be in a position absolute? it seems according to the Figma that the CircularProgress should be above the text in the loading state, while the content is centered.

Reply:
Good catch. Updated.